### ISSUE TITLE
Input parameters and outputs from training my own model

### ISSUE THREAD

**Issue by 7rebor** [cabooster/DeepCAD#2]
Hi cabooster,

I think I have successfully trained a model on my data, which is 23,075 frames of continuous 15 Hz imaging. I have a few questions about the parameters going in and the outputs:

1. What does `'--select_img_num'` mean? I see it says `'select the number of images'`, is this the number of images from the dataset that will actually go in to the training (i.e. leaving some data out to test on?). I left this at 10,000 for my 23,075 frames.
2. What does `'--train_datasets_size'` mean? Is this meant to be the total number of frames in my training dataset? I left this at 1,200, but I have 23,075 frames, was this incorrect?
3. The files going in the `'--output_dir'` are `.tif` files - what are these? It looks like an image of 'noise' and 'output' at the end of each training epoch?
4. The files going in to the `'--pth_path'` are `.pth `or `.yaml`. For your example training there was only a single `.pth` file, 'G_12_1199.pth'. Is each `.pth` file a model from each epoch of training? Is it best to use the model from the final epoch of training to test on new data?
5. Linking these questions, should I be looking for the best 'output.tif' file from all the epochs and then using that epoch's .pth file in future testing?
6. Is `'--normalize_factor'` related to the bit-depth of the image? I see it is 65535 by default, is this because you were working with 16-bit images?

Many thanks in advance for your answers,

Rob

**Comment by cabooster** [comment]
1.‘select_img_num‘ means exactly what you say. Because our program loads the image stack (the length of the image stack is set by ‘select_img_num’) in memory during the training process, we set this parameter to prevent it from running out of computer memory.
2.’train_datasets_size’ has nothing to do with the length of the entire image sequence. Our training process should divide the load image stack into small image stacks (64x64x400 or another size by your set). This parameter determines the number of small image stacks used in the training process.
3.Our program saves some intermediate results of the training process. When I first started this project, I debugged the code based on these results. You can ignore these results when using the algorithm.
4.You'd better test each model of different epochs to select the best model. Because too many training epochs are prone to overfitting, we need to choose the best model.
5.Yes, it is necessary.
6.Yes, it depends on the image bit depth.

Thank you for your questions, we will continue to optimize our code.

**Comment by 7rebor** [comment]
Thank you for your answers, they are really clearing things up for me!
